### PR TITLE
add 'clusterResourceQuota' field in NSTemplateTier

### PR DESCRIPTION
**Pull request by xcoulon**
## Description
This change will allow for configuring cluster resource quotas
in the user namespaces created on the member clusters

## Checks
1. Have you run `make generate` target? **yes**

2. Does `make generate` change anything in other projects (host-operator, member-operator, toolchain-common)? **yes** 

3. In case other projects are changed, please provides PR links.
    - host-operator: https://github.com/codeready-toolchain/host-operator/pull/156

Updates https://issues.redhat.com/browse/CRT-459
Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go [hunk 1]
```diff
@@ -31,6 +34,16 @@ type NSTemplateTierNamespace struct {
 	Template templatev1.Template `json:"template"`
 }
 
+// NSTemplateTierClusterResources defines the cluster-scoped resources associated with a given user
+type NSTemplateTierClusterResources struct {
+
+	// The revision of the corresponding template
+	Revision string `json:"revision"`
+
+	// Template contains an OpenShift Template to be used for the cluster resource quota provisioning
```
Comment (MatousJobanek) [inline pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go]:
```suggestion
	// Template contains an OpenShift Template to be used for provisioning of cluster-scoped resources
```

File: pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go [hunk 2]
```diff
@@ -16,6 +16,9 @@ type NSTemplateTierSpec struct {
 	// The namespace templates
 	// +listType
 	Namespaces []NSTemplateTierNamespace `json:"namespaces"`
+
+	// the cluster resources template (for cluster-wide quotas, etc.)
+	ClusterResources NSTemplateTierClusterResources `json:"clusterResources"`
```
Comment (alexeykazakov) [inline pkg/apis/toolchain/v1alpha1/nstemplatetier_types.go]: Let's make it optional.
```suggestion
	// the cluster resources template (for cluster-wide quotas, etc.)
	// +optional
	ClusterResources NSTemplateTierClusterResources `json:"clusterResources,omitempty"`
```

File: pkg/apis/toolchain/v1alpha1/zz_generated.openapi.go [hunk 3]
```diff
@@ -577,12 +577,18 @@ func schema_pkg_apis_toolchain_v1alpha1_NSTemplateTierSpec(ref common.ReferenceC
...
 					},
+					"clusterResources": {
+						SchemaProps: spec.SchemaProps{
+							Description: "the cluster resources template (for cluster-wide quotas, etc.)",
+							Ref:         ref("github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1.NSTemplateTierClusterResources"),
+						},
+					},
 				},
-				Required: []string{"namespaces"},
+				Required: []string{"namespaces", "clusterResources"},
```
Comment (MatousJobanek) [inline pkg/apis/toolchain/v1alpha1/zz_generated.openapi.go]: according to the latest changes, the `clusterResources` should be optional

Comment (xcoulon) [reply pkg/apis/toolchain/v1alpha1/zz_generated.openapi.go]: fixed in https://github.com/codeready-toolchain/api/pull/116/commits/e46c7c163ff6d4b5a24743305d598d3acac97d5f

**Pull request merged**